quickstarts/microsoft.containerinstance/aci-sftp-files/main.bicep

```diff
@@ -0,0 +1,142 @@
+@description('Storage account type')
+@allowed([
+  'Standard_LRS'
+  'Standard_GRS'
+])
+param storageAccountType string = 'Standard_LRS'
+
+@description('Name of file share to be created')
+param fileShareName string = 'sftpfileshare'
+
+@description('Username to use for SFTP access')
+param sftpUser string
+
+@description('Password to use for SFTP access')
+@secure()
+param sftpPassword string
+
+@description('Primary location for resources')
+param location string = resourceGroup().location
+
+var scriptName = 'createFileShare'
+var identityName = 'scratch'
+var roleDefinitionId = resourceId('Microsoft.Authorization/roleDefinitions', 'b24988ac-6180-42a0-ab88-20f7382dd24c')
+var roleAssignmentName = guid(identityName, roleDefinitionId)
+var sftpContainerName = 'sftp'
+var sftpContainerGroupName = 'sftp-group'
+var sftpContainerImage = 'atmoz/sftp:latest'
+var sftpEnvVariable = '${sftpUser}:${sftpPassword}:1001'
+var storageAccountName = 'sftpstg${uniqueString(resourceGroup().id)}'
+
+resource managedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = {
+  name: identityName
+  location: location
+}
+
+resource roleAssignment 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = {
```

Comment:
this should be scoped to the storageAccount since it appears those are the only perms needed

Reply:
@bmoore-msft So I did this:

resource roleAssignment 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = {
  name: roleAssignmentName
  scope: storageAccount <<<<<<<<<<<
  properties: {
    roleDefinitionId: roleDefinitionId
    principalId: managedIdentity.properties.principalId
    principalType: 'ServicePrincipal'
  }
}

and removed the dependson roleAssignment from storageAccount.  Is that what you mean?